src/pages/Main/index.tsx

```diff
@@ -1,68 +1,69 @@
-`
-
-const Main = () => {
-  const [keyword, setKeyword] = useState<string>('')
-  const inputRef = useRef<HTMLInputElement>(null)
-
-  const updateValue = () => {
-    if (inputRef.current) {
-      if (!inputRef.current.value.trim()) {
-        alert('검색어를 입력해주세요.')
-        inputRef.current.value = ''
-      } else {
-        setKeyword(inputRef.current.value)
-      }
-    }
-
-    // if (inputRef.current && inputRef?.current.value.trim()) {
-    //   setKeyword(inputRef.current.value)
-    //   // inputRef.current.value = ''
-    // } else {
-    //   alert('검색어를 입력해주세요.')
-    // }
-  }
-
-  const onKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
-    if (e.key === 'Enter') updateValue()
-  }
-
-  return (
-    <Wrapper>
-      <ListView>
-        <ListTop>
-          <SearchBox>
-            <input
-              type="text"
-              ref={inputRef}
-              placeholder="편의점을 검색하세요."
-              onKeyUp={onKeyUp}
-            />
-            <button onClick={updateValue}>
-              <SearchOutlined />
-            </button>
-          </SearchBox>
-
-          <FilterBtn>
-            <FilterOutlined />
-          </FilterBtn>
-        </ListTop>
-
-        <ListBox />
-      </ListView>
-
-      <MapContainer keyword={keyword} />
-    </Wrapper>
-  )
-}
-
-export default Main
+import React, { useRef, useState } from 'react'
+import MapContainer from '@components/MapContainer/MapContainer'
+import { SearchOutlined, FilterOutlined } from '@ant-design/icons'
+import styled from 'styled-components'
+import { Wrapper, ListView, ListTop, SearchBox, SortBtn } from './Main.styles'
+import ListBox from '@components/ListBox/ListBox'
+import FilterBox from '@components/FilterBox/FilterBox'
+
+styled(MapContainer)`
+  width: 70vw;
+  border: 1px solid #222;
+`
+
+const Map = () => {
+  const [keyword, setKeyword] = useState<string>('')
+  const [isFiltering, setIsFiltering] = useState(false)
+  const sortBtnRef = useRef<HTMLButtonElement | null>(null)
+  const inputRef = useRef<HTMLInputElement | null>(null)
+
+  const updateValue = () => {
+    const { current } = inputRef
+
+    if (!current) return
+    if (!current.value.trim()) {
+      alert('검색어를 입력해주세요.')
+      current.value = ''
+    } else {
+      setKeyword(current.value)
+    }
+  }
+
+  return (
+    <Wrapper>
+      <ListView>
+        <ListTop>
+          <SearchBox>
+            <input
+              type="text"
+              ref={inputRef}
+              placeholder="편의점을 검색하세요."
+              onKeyUp={(e) => {
+                if (e.key === 'Enter') updateValue()
+              }}
+            />
+            <button onClick={updateValue}>
+              <SearchOutlined />
+            </button>
+          </SearchBox>
+
+          <SortBtn
+            ref={sortBtnRef}
+            onClick={() => {
+              setIsFiltering((prev) => !prev)
+              sortBtnRef.current?.classList.toggle('on')
+            }}
+          >
```

Comment:
(p4) `sortBtnRef.current?.classList.toggle('on')` 직접 DOM 을 조작하는 부분은 지양하는게 좋을 것 같습니다
아래와 같은 방법은 안되나요?

```suggestion
          <SortBtn
            ref={sortBtnRef}
            className={isFiltering ? 'on' : ''}
            onClick={() => {
              setIsFiltering((prev) => !prev)
            }}
          >
```